### PR TITLE
RSFB-523 TO_TIME() and TIME() function support

### PR DESCRIPTION
**Pull request by akashshukla-dm**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 1]
```diff
@@ -780,6 +780,41 @@ private SqlLibraryOperators() {
...
+   * The "TO_TIME(string1, string2)" function; casts string1
+   * to a TIME using the format specified in string2.
+   */
+  @LibraryOperator(libraries = {SNOWFLAKE})
+  public static final SqlFunction TO_TIME =
+      new SqlFunction("TO_TIME",
+          SqlKind.OTHER_FUNCTION,
+          ReturnTypes.DATE_NULLABLE,
+          null,
+          OperandTypes.STRING_STRING,
```
Comment (vinayujeedm) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java:793]: Supported operands types for to_time in SF are OperandType.ANY and OperandType.STRING where second operand is optional. 
Do the changes accordingly.

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 2]
```diff
@@ -780,6 +780,41 @@ private SqlLibraryOperators() {
...
+          SqlFunctionCategory.TIMEDATE);
+
+  /**
+   * The "TIME(string1)" function; casts string1
+   * to a TIME using the format specified in string2.
+   */
+  @LibraryOperator(libraries = {BIG_QUERY})
+  public static final SqlFunction TIME =
```
Comment (vinayujeedm) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java:801]: Check previous comment for TIME and TIMESTAMP as well.

**Review by vinayujeedm — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java (hunk 2 above)
Comment (vikassuryawanshi1) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java:801]: corrected

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 3]
```diff
@@ -780,6 +780,41 @@ private SqlLibraryOperators() {
...
 
+  /**
+   * The "TO_TIME(string1, string2)" function; casts string1
+   * to a TIME using the format specified in string2.
+   */
+  @LibraryOperator(libraries = {SNOWFLAKE})
+  public static final SqlFunction TO_TIME =
+      new SqlFunction("TO_TIME",
+          SqlKind.OTHER_FUNCTION,
+          ReturnTypes.DATE_NULLABLE,
```
[the remark this answers is not in the document]
Comment (vikassuryawanshi1) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java]: corrected

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java (hunk 1 above)
Comment (vikassuryawanshi1) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java]: corrected

**Review by dm-shantanuc — APPROVED** [review]
Changes look good.

**Review by vinayujeedm — APPROVED** [review]

**Pull request merged**